pkg/dependency/parser/golang/binary/parse.go

```diff
@@ -145,11 +146,34 @@ func (p *Parser) ldFlags(settings []debug.BuildSetting) []string {
 			continue
 		}
 
-		return strings.Fields(setting.Value)
+		flags, err := splitLDFlags(setting.Value)
+		if err != nil {
+			p.logger.Error("Could not parse -ldflags found in build info", log.Err(err))
+			return nil
+		}
+
+		return flags
 	}
 	return nil
 }
 
+// splitLDFlags separates flags with spaces.
+// If a flag contains nested flags - the nested flags will be stored as a single element
+func splitLDFlags(flags string) ([]string, error) {
+	allFlags, err := shlex.Split(flags)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to split flags: %w", err)
+	}
+
+	for i, flag := range allFlags {
+		ss := strings.Split(flag, "=")
+		ss[0] = strings.ReplaceAll(ss[0], "h", "") // remove `h` from flags to avoid help flag error.
+		allFlags[i] = strings.Join(ss, "=")
+	}
```

Comment:
I changed logic in https://github.com/aquasecurity/trivy/pull/8368/commits/06bb746dfeafe2f00702ff38a34290fac2208ac1 as suggested @knqyf263 (see https://github.com/aquasecurity/trivy/issues/8365#issuecomment-2644528317)

Reply:
I found some time to think about it and got an idea.
https://github.com/aquasecurity/trivy/pull/8368#issuecomment-2653341325

Reply:
Updated in https://github.com/aquasecurity/trivy/pull/8368/commits/7ef23b48d7a397e93d33c082ed71009287c2dcd5